packages/php-wasm/universal/src/lib/base-php.ts

```diff
@@ -616,6 +616,39 @@ export abstract class BasePHP implements IsomorphicLocalPHP {
 		this[__private__dont__use].FS.rename(fromPath, toPath);
 	}
 
+	/** @inheritDoc */
+	@rethrowFileSystemError('Could not copy "{path}"')
+	cp(fromPath: string, toPath: string) {
+		const FS = this[__private__dont__use].FS;
+
+		const fromStat = FS.stat(fromPath);
+
+		const fromMode = parseInt(fromStat.mode.toString(8).substring(0, 1));
+
+		const fromIsDir = fromMode === 4;
+
+		if (fromIsDir) {
+			try {
+				FS.mkdir(toPath);
+			} catch {
+				// Do nothing
+			}
+
+			const files = FS.readdir(fromPath);
+
+			files
+				.filter((f: string) => !['.', '..'].includes(f))
+				.forEach((f: string) =>
+					this.cp(fromPath + '/' + f, toPath + '/' + f)
+				);
+
+			FS.rmdir(fromPath);
```

Comment:
Why not `this.rmdir`? It's defined just a few lines below

Reply:
@adamziel That's been refactored out.